doc/architecture/decisions/0005-rooms-and-beds-for-both-cas-services.md

```diff
@@ -0,0 +1,106 @@
+         `---------------------------------------'
+                 |                |
+                 |                |----------`
+                 |                          .|.
+                 |           ,---------------------------.
+                 |           |PremiseCharacteristic      |
+                 |           |---------------------------|
+                 |           |*premise_id: integer       |
+                 |           |*characteristic_id: integer|
+                 |           `---------------------------'
+                 |                           .|.
+                 |                            |
+                 |                            |
+                .|.                  ,---------------------------.
+      ,---------------------------.  |Premise                    |
+      |RoomCharacteristic         |  |---------------------------|
+      |---------------------------|  |*id: integer <<autonumber>>|
+      |*room_id: integer          |  |--                         |
+      |*characteristic_id: integer|  |address: string            |
+      `---------------------------'  |postcode: string           |
+             .|.                     `---------------------------'
+              |-------`                      |
+                     |             `----------
+                     |             .|.
+               ,---------------------------.
+               |Room                       |
+               |---------------------------|
+               |*id: integer <<autonumber>>|
+               |--                         |
+               `---------------------------'
+                              |
+                             .|.
+               ,---------------------------.
+               |Bed                        |
+               |---------------------------|
+               |*id: integer <<autonumber>>|
+               |--                         |
+               `---------------------------'
+
+[2] [PlantUML](https://www.plantuml.com/plantuml/uml/fP5VQiCm3CRVVGhJnu8l44eftGN6Bb0yMsl5OYj8og0qlVkIT0dDC2omn-Zxau-V3wAUjKsraK_CZMUVR8qPzDg09TOB6GDPKQ_svTAhGO5H58Ez6Mkt62kBMU4CR7SxtohbDhq3x_UzwjnWoJv1PKKPytdu5k0l372IGCqizE487xwjjRARxMQCRnD5gF_otQTvRqIfD97x--bfvt0B2zz2dVjU6H8Ab2gRX_IFrhNqV8mC8a_y6X8D56SYIm3hUkV84UBQXFa5ekkSerk1UjbOncbCuTNkv5pNZSrcJfKSFwAPNEElkiUciNWw726Q0-JOfleR)
+
+## Consequences
+
+- For CAS3, the new 'create a Room' endpoint will also need to create a bed.
+  The implementation for this will need to be considered to allow CAS1 to do
+  this across two different transactions
+- Splitting out characteristics into a new table, potentially with join tables,
+  will require more set up work but this cost should be reasonable
+- Admin may in future need to modify the list of characteristics, having these
+  stored in a table rather than hardcoded should enable that
+- CAS1 will have to maintain their temporary `Premises.bed_count` mechanism for
```

Comment:
To allow CAS1 to continue with a per-Premises bed count but have CAS3 do bed-level modelling, Bookings and Lost Beds will need to have optional references to Beds with some switching around Premises type in various places:

- GET /premises & GET /premises/{id} (`availableBedsForToday` calculation for this will be slightly different I think as would need a Lost Bed entry for each individual Bed?)
- GET /premises/{premises id}/capacity (similar to above)
- POST /premises/{premises id}/booking (validate that a bed reference has/has-not been provided based on the type of premises.  Validate that you're not attempting to double book a bed?)
- POST /premises/{premises id}/booking/{booking id}/extensions (there's some potential complexity here if you need to extend someone's stay if the bed has already been booked for someone else after the current length of the booking - could probably just return an error in these cases and force the Probation Practitioner to make a new booking)

Might be good to make a note of some of these cases here and how the tables affected/logic will change?

Reply:
Thanks for this context, we will definitely need to plan for these details and beleive that you and Ed have just run through them.